artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -415,8 +418,16 @@ public QueueImpl(final QueueConfiguration queueConfiguration,
       scheduledDeliveryHandler = new ScheduledDeliveryHandlerImpl(scheduledExecutor, this);
 
       if (addressSettingsRepository != null) {
-         addressSettingsRepositoryListener = new AddressSettingsRepositoryListener(addressSettingsRepository);
-         addressSettingsRepository.registerListener(addressSettingsRepositoryListener);
+         this.addressSettingsRepository = addressSettingsRepository;
+         addressSettingsRepositoryListener = new NamedHierarchicalRepositoryChangeListener(queueConfiguration.getName()) {
```

Comment:
A name prefix could help to identify the listener owner and avoid conflicts between retroactive addresses and queues with the same name.

Reply:
The name used by the listener for the retroactive use-case already has a prefix, etc. so that it is unique.